sceptre/template.py

```diff
@@ -116,82 +80,22 @@ def body(self):
         :rtype: str
         """
         if self._body is None:
-            file_extension = os.path.splitext(self.path)[1]
-
-            try:
-                if file_extension in {".json", ".yaml", ".template"}:
-                    with open(self.path) as template_file:
-                        self._body = template_file.read()
-                elif file_extension == ".j2":
-                    self._body = self._render_jinja_template(
-                        os.path.dirname(self.path),
-                        os.path.basename(self.path),
-                        {"sceptre_user_data": self.sceptre_user_data}
-                    )
-                elif file_extension == ".py":
-                    self._body = self._call_sceptre_handler()
-
-                else:
-                    raise UnsupportedTemplateFileTypeError(
-                        "Template has file extension %s. Only .py, .yaml, "
-                        ".template, .json and .j2 are supported.",
-                        os.path.splitext(self.path)[1]
-                    )
-            except Exception as e:
-                self._print_template_traceback()
-                raise e
+            type = self.handler_config.get("type")
+            handler_class = self._get_handler_of_type(type)
+            handler = handler_class(
+                name=self.name,
+                arguments={k: v for k, v in self.handler_config.items() if k != "type"},
```

Comment:
we exclude it because we do not need to validate the `type` at this point.  It gets validated when loading the template template handlers in the `_get_handler_of_type` function.